src/Icicle/Avalanche/ToJava.hs

```diff
@@ -280,10 +283,15 @@ primTypeOfPrim p
   dble     M.PrimDoubleExp  = Function "Math.exp"
   dble     M.PrimDoubleSqrt = Function "Math.sqrt"
 
-  cast      M.PrimCastIntOfDouble = Function "(int)"
-  cast      M.PrimCastDoubleOfInt = Function "(double)"
-  cast      M.PrimCastStringOfInt = Function "Integer.toString"
-  cast      M.PrimCastStringOfDouble = Function "Double.toString"
+  toInt    M.PrimToIntFloor      = Function "Math.floor"
+  toInt    M.PrimToIntCeiling    = Function "Math.ceil"
+  toInt    M.PrimToIntRound      = Function "Math.round"
+  toInt    M.PrimToIntTruncate   = Function "Math.trunc"
```

Comment:
the SO answer was something along the lines of a DecimalFormatter object
![image](https://cloud.githubusercontent.com/assets/235039/12498651/43c57e1e-c0f8-11e5-9fd8-8d60de42d439.png)